qiskit/pulse/parser.py

```diff
@@ -162,7 +165,10 @@ def visit_Expression(self, node):
         Returns:
             ast.Expression: Evaluated value.
         """
-        return self.visit(node.body)
+        tmp_node = copy.deepcopy(node)
```

Comment:
Could you please help me understand why copying is necessary now?

Reply:
Because it overwrites original ast node, we need to copy it before calling `visit`. `tmp_node` is returned and passed to create new parser object if parameters are partially bound. Otherwise, we cannot reuse a parser for a different parameter set.

Reply:
I wrote this test to check if the parser can be reused for another parameter.
https://github.com/Qiskit/qiskit-terra/blob/b0d9b0b247987b9b6356a07768ef7d4ee5e1e0cc/test/python/pulse/test_parser.py#L207-L215